### ISSUE TITLE
net: there should be a net.ResolveAddr() function

### ISSUE THREAD

**Issue by siebenmann** [golang/go#5133]
<pre>Now that there is net.DialOpt() and net.LocalAddress(), it would be quite
convenient to have a net.ResolveAddr() function so that you can generically
create the local address Addr that net.LocalAddress() wants. Right now
you have to write code that switches on the network type to call the
appropriate Resolve&lt;whatever&gt;Addr() function; among other issues, this
burns knowledge of what network types the net package supports into your
own code.

Basically all of the code necessary already exists in the internal
net resolveAddr() function; all that would be necessary would be
exposing a public interface to it.</pre>

**Comment by bradfitz** [comment]
Does net.Resolver in Go 1.8 solve this?

If not, please file a new bug about what's tedious to do with Go 1.8 (with sample code) and reference this bug in your new bug report.
